packages/okta-react-native/index.test.js

```diff
@@ -0,0 +1,349 @@
+        getAccessToken: jest.fn(),
+        getIdToken: jest.fn(),
+        getUser: jest.fn(),
+        isAuthenticated: jest.fn(),
+        revokeAccessToken: jest.fn(),
+        revokeIdToken: jest.fn(),
+        revokeRefreshToken: jest.fn(),
+        introspectAccessToken: jest.fn(),
+        introspectIdToken: jest.fn(),
+        introspectRefreshToken: jest.fn(),
+        refreshTokens: jest.fn(),
+      },
+    },
+    Platform: {
+      OS: 'ios',
+      select: jest.fn()
+    },
+    NativeEventEmitter: jest.fn(),
+    DeviceEventEmitter: jest.fn(),
+  });
+});
+
+jest.mock('NativeEventEmitter');
+
+describe('OktaReactNative', () => {
+
+  beforeAll(() => {
+  });
+
+  describe('createConfigTest', () => {
+    let mockCreateConfig;
+
+    const config = {
+      clientId: 'dummy_client_id',
+      redirectUri: 'dummy://redirect', 
+      endSessionRedirectUri: 'dummy://endSessionRedirect', 
+      discoveryUri: 'https://dummy_issuer',
+      scopes: ['scope1'],
+      requireHardwareBackedKeyStore: true
+    };
+
+    beforeEach(() => {
+      mockCreateConfig = require('react-native').NativeModules.OktaSdkBridge.createConfig;
+    });
+    
+    it('passes in correct parameters on ios device', () => {
+      Platform.OS = 'ios';
+      const processedScope = config.scopes.join(' ');
+      createConfig(config);
+      expect(mockCreateConfig).toHaveBeenCalledWith(
+        config.clientId,
+        config.redirectUri,
+        config.endSessionRedirectUri,
+        config.discoveryUri,
+        processedScope,
+      );
+    });
+
+    it('passes in correct parameters on android device', () => {
+      Platform.OS = 'android';
+      createConfig(config);
+      expect(mockCreateConfig).toHaveBeenCalledWith(
+        config.clientId,
+        config.redirectUri,
+        config.endSessionRedirectUri,
+        config.discoveryUri,
+        config.scopes,
+        config.requireHardwareBackedKeyStore,
+      );
+    });
+  });
+
+  describe('signInTest', () => {
+    let mockSignIn;
+
+    beforeEach(() => {
+      mockSignIn = require('react-native').NativeModules.OktaSdkBridge.signIn;
+    });
+
+    it('calls native sign in method', () => {
+      signIn();
+      expect(mockSignIn).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe('signOutTest', () => {
+    let mockSignOut;
+
+    beforeEach(() => {
+      mockSignOut = require('react-native').NativeModules.OktaSdkBridge.signOut;
+    });
+
+    it('calls native sign out method', () => {
+      signOut();
+      expect(mockSignOut).toHaveBeenCalledTimes(1);
+    });
+  });
+  
+  describe('getAccessTokenTest', () => {
+    let mockGetAccessToken;
+
+    beforeEach(() => {
+      mockGetAccessToken = require('react-native').NativeModules.OktaSdkBridge.getAccessToken;
+    });
+
+    it('gets access token successfully', async () => {
+      mockGetAccessToken.mockReturnValueOnce('dummy_access_token');
+      expect.assertions(1);
+      const data = await getAccessToken();
```

Comment:
What happens if getAccessToken() fails/throws?

Reply:
i can mock the modules to throw errors, but those will be mocked :)

Reply:
It's more of a check to see - have you made these tests fail?   More than once I've had tests that looked good, but it later turned out that a failure was getting trapped and I never knew and the test turned out to be worthless.  

If you've checked that case, there's no change needed here, but it's something to consider.

Reply:
@swiftone I've made the the method fail, but since `getAccessToken()` is a one line method: `return NativeModules.OktaSdkBridge.getAccessToken();`, and it's accessing the native module asynchronously (I'm mocking the native module also). I don't really know how to test the case which it fails :p, unless I mock the return val of the method to throw an error. A lot of the methods being tested here are one line method that calls the native modules.

Reply:
I like to think of the tests as documenting the API.

So here, if the source native module `getAccessToken()` throws, you expect your wrapper to throw that same error.  You can document (test) this by mocking the source to throw an error, and have a test that traps the error to confirm it was thrown.

Then anyone reading your tests immediately sees what is expected in case of an error, without digging into the code.

I'll approve here because that level of test isn't NEEDED...but it's a good idea to consider it if you have a chance.

Reply:
I'll definitely keep this in mind! Will update the tests with another PR if I end up with time for this.